Update release guide instructions post v3.7.0 (#25235)
matplotlib/matplotlib: doc/devel/release_guide.rst

   ``remote`` and a read/write remote is ``DANGER``

.. _release_feature_freeze:

Making the release branch
=========================

When a new minor release (vX.Y.0) is approaching, a new release branch must be made.

> When precisely this happens is up to the release manager, but at this point is where

Reviewer: ```suggestion
When precisely this should happen is up to the release manager, but this point is where
```

most new features intended for the minor release are merged and you are entering a
feature freeze (i.e. newly implemented features will be going into vX.Y+1).
This does not necessarily mean that no further changes will be made prior to release,
just that those changes will be made using the backport system.

> First create the branch ::

Reviewer: ```suggestion
For an upcoming ``v3.7.0`` release, first create the branch::
```

   python tools/memleak.py agg 1000 agg.pdf

Run the User Acceptance Tests for the NBAgg and ipympl backends::

   jupyter notebook lib/matplotlib/backends/web_backend/nbagg_uat.ipynb

For ipympl, restart the kernel, add a cell for ``%matplotlib widget`` and do
not run the cell with ``matplotlib.use('nbagg')``. Tests which check

> ``connection_info``, use ``reshow``, or test the OO interfeace are not expected

Reviewer: ```suggestion
``connection_info``, use ``reshow``, or test the OO interface are not expected
```

   git branch v2.0.0-doc
   git push DANGER v2.0.0-doc

Update (or create) the ``v2.0-doc`` milestone.
The description should include the instruction for meeseeksmachine to backport changes
with the ``v2.0-doc`` milestone to both the ``v2.0.x`` branch and the ``v2.0.0-doc`` branch::

   Documentation changes (.rst files and examples)

   on-merge: backport to v2.0.x
   on-merge: backport to v2.0.0-doc

Check all active milestones for consistency. Older doc milestones should also backport to
higher minor versions (e.g. ``v3.6-doc`` should backport to both ``v3.6.x`` and ``v3.7.x``
if the ``v3.7.x`` branch exists)

On this branch un-comment the globs from :ref:`release_chkdocs`.  And then ::

>    git push DANGER v2.1.x

Reviewer: I'm confused why you're pushing `v2.1.x` when it's only been created in the next step?

Author: I'm actively reworking this, I knew it was not ready (I had a discussion with Tom, which came to the same conclusions as these comments, just hadn't actually written out the updated text yet)

If this is the last micro release anticipated (or otherwise are entering feature
freeze for the next minor release), create a release branch for the next minor
release ::

   git switch main

>    git branch v2.1.x

Reviewer: This is not normally something we do. Everything goes to `main` until we decide to branch near-ish the release candidate. It's not related to the tagging of the last bugfix/micro.

It typically takes about 5-10 minutes for the website to process the push and update the
live web page (remember to clear your browser cache).

.. _release_merge_up:

Merge up changes to main
========================

After a release is done, the changes from the release branch should be merged into the
``main`` branch. This is primarily done so that the released tag is on the main branch
so ``git describe`` (and thus ``setuptools-scm``) has the most current tag.
Secondarily, changes made during release (including removing individualized release
notes, fixing broken links, and updating the version switcher) are bubbled up to
``main``.

Git conflicts are very likely to arise, though aside from changes made directly to the

> release branch (mostly as part of the release), should be relatively trivially resolved

Reviewer: ```suggestion
release branch (mostly as part of the release), they should be relatively trivially resolved
```

It typically takes about 5-10 minutes for the website to process the push and update the
live web page (remember to clear your browser cache).

.. _release_merge_up:

Merge up changes to main
========================

After a release is done, the changes from the release branch should be merged into the
``main`` branch. This is primarily done so that the released tag is on the main branch
so ``git describe`` (and thus ``setuptools-scm``) has the most current tag.
Secondarily, changes made during release (including removing individualized release
notes, fixing broken links, and updating the version switcher) are bubbled up to
``main``.

Git conflicts are very likely to arise, though aside from changes made directly to the

> release branch (mostly as part of the release), they should be relatively trivially resolved

Reviewer: ```suggestion
release branch (mostly as part of the release), they should be relatively-easily resolved
```
